firebase-vscode/src/core/project.ts

```diff
@@ -42,17 +59,57 @@ export function registerProject(broker: ExtensionBrokerImpl): Disposable {
     });
   });
 
+  // Update .firebaserc with defined project ID
+  effect(() => {
+    const projectId = currentProjectId.value;
+    if (projectId) {
+      updateFirebaseRCProject(context, "default", currentProjectId.value);
+    }
+  });
+
   broker.on("getInitialData", () => {
     broker.send("notifyProjectChanged", {
       projectId: currentProject.value?.projectId ?? "",
     });
   });
 
   broker.on("selectProject", async () => {
-    // TODO: implement at the same time we teardown the old picker
-    // const projects = await listProjects();
-    // const id = await promptUserForProject(projects);
-    // pluginLogger.info("foo:", { id });
+    if (process.env.MONOSPACE_ENV) {
+      pluginLogger.debug(
+        "selectProject: found MONOSPACE_ENV, " +
+          "prompting user using external flow"
+      );
+      /**
+       * Monospace case: use Monospace flow
+       */
+      const monospaceExtension =
+        vscode.extensions.getExtension("google.monospace");
+      process.env.MONOSPACE_DAEMON_PORT =
+        monospaceExtension.exports.getMonospaceDaemonPort();
+      try {
+        const projectId = await selectProjectInMonospace({
+          projectRoot: currentOptions.cwd,
+          project: undefined,
+          isVSCE: true,
+        });
+
+        if (projectId) {
+          currentProjectId.value = projectId;
+        }
+      } catch (e) {
+        pluginLogger.error(e);
+      }
+    } else if (isServiceAccount.value) {
+      return;
```

Comment:
Can this check be moved up? To avoid unnecessary async call?

Reply:
Which async call? I don't want to make too many logic changes since I'm not that familiar w/ monospace.